tuf/ngclient/updater.py

```diff
@@ -122,12 +126,38 @@ def refresh(self) -> None:
             RepositoryError: Metadata failed to verify in some way
             DownloadError: Download of a metadata file failed in some way
         """
+        #                                     json.load()
+        # Load self.spec_version using self._load_local_metadata() here from disk
+        # Exception would be raised if spec_version is not found locally. Set to "1" in that case
+        repository_versions = self._get_repository_versions()
+
+        # Updating self.spec_version
+        specver_message = _get_spec_version(
+            repository_versions, self.spec_version, SUPPORTED_VERSIONS
+        )
+
+        if specver_message(1):
+            logger.warning(specver_message(1))
+        self.spec_version = specver_message(0)
+        self._persist_metadata("spec_version", json.dumps(self.spec_version))
 
         self._load_root()
         self._load_timestamp()
         self._load_snapshot()
         self._load_targets(Targets.type, Root.type)
 
+    def _get_repository_versions(self) -> List[str]:
+        """Returns a list of all the repository versions."""
+
+        url = f"{self._metadata_base_url}supported-versions.json"
+
+        with self._fetcher.download_file(
+            url, "length placeholder"
```

Comment:
probably want `self.config.supported_versions_max_length` in place of `"length placeholder"`. Need to add a line to `tuf/ngclient/config.py` as well.

Reply:
Considering my comment above, should `supported_versions_max_length` have the length in bytes equivalent?

Reply:
Yes, it should use 'bytes' as the unit

Reply:
What could be a possible value for `supported_versions_max_length` in that case?

Reply:
Think about an example `supported-versions.json` file:

```json
{"supported_versions": ["1", "2", "3", "4"]}
```

That's got 45 bytes in it. So let's maybe round up to 1000 bytes? That's a pretty huge margin of error but well short of anything that could cause issues

Reply:
What could be the approx len() of the list in case of 1000 bytes?

Reply:
Well, a list with 1000 elements would definitely serialize to have > 1000 bytes. The real threshold would probably be a couple hundred. 

Why do you ask?